src/IntervalAwareTrait.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace RebelCode\Time;
+
+use Dhii\Util\String\StringableInterface as Stringable;
+use Exception as RootException;
+use InvalidArgumentException;
+
+/**
+ * Common functionality for objects that are aware of an interval.
+ *
+ * @since [*next-version*]
+ */
+trait IntervalAwareTrait
+{
+    /**
+     * The interval instance.
+     *
+     * @since [*next-version*]
+     *
+     * @var IntervalInterface
+     */
+    protected $interval;
+
+    /**
+     * Retrieves the interval associated with this instance.
+     *
+     * @since [*next-version*]
+     *
+     * @return IntervalInterface|null
+     */
+    protected function _getInterval()
+    {
+        return $this->interval;
+    }
+
+    /**
+     * Sets the interval for this instance.
+     *
+     * @since [*next-version*]
+     *
+     * @param IntervalInterface|null $interval
+     */
+    protected function _setInterval($interval)
+    {
+        if ($interval !== null && !($interval instanceof IntervalInterface)) {
+            throw $this->_createInvalidArgumentException(
+                $this->__('Argument is not a valid interval instance or null.'),
```

Comment:
Here and everywhere else, something like this is enough:

> Argument is not a valid interval.

Reply:
And no period, right? I recall agreeing on no period at the end.

Reply:
That can be a new convention, then: if it's just one sentence, no period.